firebase/src/core/data/gateways/firestore-collection-name.ts

```diff
@@ -0,0 +1,22 @@
+/** Typed firestore collections. */
+export type FirestoreCollectionName =
+  | PublicCollectionName
+  | PublicSubCollectionName
+  | UserCollectionName
+  | UserSubCollectionName;
+
+/** Public collections available to all authenticated users. */
+type PublicCollectionName = "public_collections" | "public_collection_categories" | "public_memos";
+
+/** Public subcollections available to all authenticated users. */
+type PublicSubCollectionName = `public_collections/${string}/public_memos`;
```

Comment:
Even this one, `public_memos` that are placed in an already "public" collection (as its name suggests)

Reply:
That's because it groups ALL firestore collections (including sub-collections) using its names.